qiime_studio/api/jobs.py

```diff
@@ -147,8 +134,8 @@ def callback(future):
             job = JOBS[job_id]
 
             if results is not None:
-                for result, path in zip(results, outputs.values()):
-                    result.save(path)
+                for key, path in outputs.items():
+                    getattr(results, key).save(path)
                 job['outputs'] = {k: v.uuid for k, v in zip(outputs, results)}
```

Comment:
actually it is probably easier to do something like `job['outputs'] = {}` and in the loop just say: 
```
for key, path in outputs.items():
    artifact = getattr(results, key)
    artifact.save(path)
    job['outputs'][key] = artifact.uuid  # (might need str here)
```